app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,25 @@
+class ReviewsController < ApplicationController
+  before_action :require_user
+
+  def create
+    @video = Video.find(params[:video_id])
+    @review = Review.new(review_params)
+    @review.video = @video
+    @review.user = current_user
+
+    if @review.save
+      flash[:message] = "Your review was created"
+      redirect_to video_path(@video)
+    else
+      @reviews = @video.reviews
+      flash[:error] = "All fields are required"
```

Comment:
Make sure you use `flash.now` when you are rendering. If you don't, your flash message will survive one more request than you expect.